### ISSUE TITLE
git-solo with no args doesn't clear duet commiter

### ISSUE THREAD

**Issue by naokiri** [git-duet/git-duet#81]
Hi, I use git-duet with GIT_DUET_CO_AUTHORED_BY so I care if `duet.env.git-committer-name` is set or not when doing solo programming.

The output from `git solo` is same as when I do `git solo ni` that I don't notice my typo, and the latter commit message is created with unwanted Co-Authored-by trailers.

What kind of fix is better for users?
For my usage I want to clear the committer when no args are given to git-solo.

Here's the example
```
$ mkdir tmp
$ cd tmp
$ git init
Initialized empty Git repository in /home/naoaki/src/tmp/.git/
$ cat ~/.git-authors 
authors:
    ni: Naoaki Iwakiri; naoaki
    hm: HHH MMM; him
    hr: HHH RRR; her
email:
  prefix:
  domain: example.com

$ git duet ni hm
GIT_AUTHOR_NAME='Naoaki Iwakiri'
GIT_AUTHOR_EMAIL='naoaki@example.com'
GIT_COMMITTER_NAME='HHH MMM'
GIT_COMMITTER_EMAIL='him@example.com'
$ git config --get-regexp 'duet.env' | cat
duet.env.git-author-initials ni
duet.env.git-author-name Naoaki Iwakiri
duet.env.git-author-email naoaki@example.com
duet.env.mtime 1568356020
duet.env.git-committer-initials hm
duet.env.git-committer-name HHH MMM
duet.env.git-committer-email him@example.com
$ git solo
GIT_AUTHOR_NAME='Naoaki Iwakiri'
GIT_AUTHOR_EMAIL='naoaki@example.com'
$ git config --get-regexp 'duet.env' | cat
duet.env.git-author-initials ni
duet.env.git-author-name Naoaki Iwakiri
duet.env.git-author-email naoaki@example.com
duet.env.mtime 1568356020
duet.env.git-committer-initials hm
duet.env.git-committer-name HHH MMM
duet.env.git-committer-email him@example.com
$ git solo ni
GIT_AUTHOR_NAME='Naoaki Iwakiri'
GIT_AUTHOR_EMAIL='naoaki@example.com'
$ git config --get-regexp 'duet.env' | cat
duet.env.git-author-initials ni
duet.env.git-author-name Naoaki Iwakiri
duet.env.git-author-email naoaki@example.com
duet.env.mtime 1568356114
duet.env.git-committer-initials 
duet.env.git-committer-name 
duet.env.git-committer-email 
```

**Comment by jszwedko** [comment]
Hi @naokiri ,

Thanks for the detailed issue and suggested fix! I can see why the current behavior could be confusing when `git duet xx yy` has been used followed by `git solo`.

Given that the current behavior of `git solo` and `git duet` is just to display, rather than modify, the current configuration, I'm a bit reticent to have it begin to modify the active pairing configuration. I'm wondering what you would think of simply having `git solo` output both parts of the pair if `git duet xx yy` was previously used. I.e. it would be an alias for `git duet` when called with no arguments.

**Comment by naokiri** [comment]
I understand the hesitance to change the behavior incompatibly. For a quick fix for current version I'm for to display both committer and author on `git solo` with no args.

But to make of the benefits having different command for solo and duet, I'd still prefer to change the behavior when giving no args to `git solo` and `git duet` both. When user types `solo` or `duet` they should be willing to do solo committing or duet committing in the close future, even when they are checking the current configuration for now. 

Edit: If we take the just to display way, then change will be like this? https://github.com/git-duet/git-duet/pull/84

**Comment by WilHall** [comment]
@jszwedko I think making `git solo` an alias for `git duet` when passed no arguments will at least make it obvious what the current configuration is. The current behavior implies that the configuration has been cleared out, which is part of the confusion.

The follow-up question I have is: What would be the use case for specifying anything other than your own initials to `git solo`? It seems like `git solo` for me implies `git solo wh` because I wouldn't solo-commit as my pair. In some setups - such as when using GPG signing - that may not even be valid.

I agree with @naokiri that `git solo` and `git duet` feel like they imply intent to enter a _mode of work_, rather than output the current state. The output is currently confusing for a few reasons:
1. As mentioned above, `git solo` masks the configured co-author (if any) implying that it cleared that config out
2. `git duet` always outputs an author and committer, even if no commiter is configured:
```
$ git solo wh
GIT_AUTHOR_NAME='Wil Hall'
GIT_AUTHOR_EMAIL='wil@wilhall.com'

$ tail ~/.gitconfig
[duet "env"]
        git-author-initials = wh
        git-author-name = Wil Hall
        git-author-email = wil@wilhall.com
        mtime = 1569078541
        git-committer-initials =
        git-committer-name =
        git-committer-email =

$ git duet
GIT_AUTHOR_NAME='Wil Hall'
GIT_AUTHOR_EMAIL='wil@wilhall.com'
GIT_COMMITTER_NAME='Wil Hall'
GIT_COMMITTER_EMAIL='wil@wilhall.com'

```

I think what would feel more natural would be `git solo` with no arguments clearing out all duet configs, and `git duet` requiring two sets of initials. Showing configuration could be a separate command (maybe something like`git [duet|solo] --show`? `git [duet|solo]-show`? `git [duet|solo]-config`)

**Comment by jszwedko** [comment]
Thanks @naokiri and @WilHall for your thoughts!

https://github.com/git-duet/git-duet/pull/84 looks good for printing the committer even in the `git solo` situation.

I am coming around to the idea that `git solo` and `git duet` should "activate" the state with a separate command or flag for printing the current state (such as the suggested `git duet --show`). Given the existing behavior of simply outputting the configuration has been the behavior since the project was created, I still hesitate to change it for compatibility reasons, but I'd be open to introducing a new environment variable that would modify the behavior of `git duet` and `git solo` for now and change the default when/if a new major version is released.

To summarize, for the altered mode, it would be:
* `git solo <initials>` activates solo-mode with the given initials
* `git solo` remove `git duet` config to use whatever is in `git config user.name` and `git config user.email`
* `git duet --show` (or similar): prints the current author and and (if set) committer info
* `git duet <initials> <initials>`: activates duet-mode with the given set of initials
* `git duet`: prints help message indicating initials are required

Does this sound right?

**Comment by naokiri** [comment]
For altered mode, yes. I think that will be nice.
Just one case missing; Is it OK to show some message like 'not a git repository' and exit with 1 for altered mode and `git solo` without `-g` option ran on non-git repository?

Also, `--show` can be not only for altered mode, I think adding `--show` option to current version is better for user who wants to prepare keeping the current behavior.

Edit: Updated #82 to implement the suggested change.
